packages/typespec-vscode/src/extension.ts

```diff
@@ -213,3 +231,16 @@ function showStartUpMessages(stateManager: ExtensionStateManager) {
     stateManager.cleanUpStartUpMessage(workspaceFolder.uri.fsPath);
   });
 }
+
+async function showOpenApi3(docUri: vscode.Uri, context: vscode.ExtensionContext) {
+  const selectedFile = docUri.fsPath;
+  const mainTspFile = selectedFile.endsWith("main.tsp") ? selectedFile : await getMainTspFile();
+  if (mainTspFile === undefined) {
+    const errMsg = `No 'main.tsp' file can be determined from '${selectedFile}' in workspace.`;
```

Comment:
you can use logger.error() with popup option.

Reply:
updated all `vscode.workspace.showErrorMessage()` to use `logger.error()` with popup options as much as possible.